{
  "errors": [],
  "generated_at": "2024-10-27T18:00:56Z",
  "metrics": {
    "D:\\bandit\\upstream-repos\\py3-python-pypi-mirror\\pypi_mirror.py": {
      "CONFIDENCE.HIGH": 3,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 756,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-python-pypi-mirror\\setup.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 26,
      "nosec": 0,
      "skipped_tests": 0
    },
    "_totals": {
      "CONFIDENCE.HIGH": 3,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 2,
      "SEVERITY.MEDIUM": 1,
      "SEVERITY.UNDEFINED": 0,
      "loc": 782,
      "nosec": 0,
      "skipped_tests": 0
    }
  },
  "results": [
    {
      "code": "16 import shutil\n17 import subprocess\n18 import sys\n",
      "col_offset": 0,
      "end_col_offset": 17,
      "filename": "D:\\bandit\\upstream-repos\\py3-python-pypi-mirror\\pypi_mirror.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Consider possible security implications associated with the subprocess module.",
      "line_number": 17,
      "line_range": [
        17
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/blacklists/blacklist_imports.html#b404-import-subprocess",
      "test_id": "B404",
      "test_name": "blacklist"
    },
    {
      "code": "276     args += pkgs\n277     subprocess.check_call(args)\n278 \n",
      "col_offset": 4,
      "end_col_offset": 31,
      "filename": "D:\\bandit\\upstream-repos\\py3-python-pypi-mirror\\pypi_mirror.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 78,
        "link": "https://cwe.mitre.org/data/definitions/78.html"
      },
      "issue_severity": "LOW",
      "issue_text": "subprocess call - check for execution of untrusted input.",
      "line_number": 277,
      "line_range": [
        277
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b603_subprocess_without_shell_equals_true.html",
      "test_id": "B603",
      "test_name": "subprocess_without_shell_equals_true"
    },
    {
      "code": "821         url = args.url.format(pkg=args.pkg[0])\n822         with urllib.request.urlopen(url) as resp:\n823             pkg_info = json.loads(resp.read().decode())\n",
      "col_offset": 13,
      "end_col_offset": 40,
      "filename": "D:\\bandit\\upstream-repos\\py3-python-pypi-mirror\\pypi_mirror.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 22,
        "link": "https://cwe.mitre.org/data/definitions/22.html"
      },
      "issue_severity": "MEDIUM",
      "issue_text": "Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected.",
      "line_number": 822,
      "line_range": [
        822
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/blacklists/blacklist_calls.html#b310-urllib-urlopen",
      "test_id": "B310",
      "test_name": "blacklist"
    }
  ]
}